src/napari_graph/interops.py

```diff
@@ -0,0 +1,109 @@
+from typing import Any
+
+import networkx as nx
+import numpy as np
+import pandas as pd
+
+from napari_graph.base_graph import BaseGraph
+from napari_graph.directed_graph import DirectedGraph
+from napari_graph.undirected_graph import UndirectedGraph
+
+
+def from_networkx(graph: nx.Graph) -> BaseGraph:
+    """Convert a NetworkX graph into a napari-graph UndirectedGraph or DirectedGraph.
+
+    Parameters
+    ----------
+    graph : nx.Graph
+        The NetworkX graph to be converted.
+
+    Returns
+    -------
+    BaseGraph
+        An equivalent napari-graph UndirectedGraph or DirectedGraph.
+    """
+
+    coords_dict = nx.get_node_attributes(graph, "pos")
+    if len(coords_dict) > 0:
+        coords_df = pd.DataFrame.from_dict(coords_dict, orient="index")
+    else:
+        coords_df = None
+
+    edges = graph.edges
+    if len(edges) > 0:
+        edges = np.atleast_2d(edges)
+
+    if graph.is_directed():
+        out_graph = DirectedGraph(edges, coords_df)
+    else:
+        out_graph = UndirectedGraph(edges, coords_df)
+
+    return out_graph
+
+
+def to_networkx(graph: BaseGraph) -> nx.Graph:
+    """Convert a napari-graph UndirectedGraph or DirectedGraph into NetworkX graph.
+
+    Parameters
+    ----------
+    graph : BaseGraph
+        napari-graph Graph
+
+    Returns
+    -------
+    nx.Graph
+        An equivalent NetworkX graph.
+    """
+
+    if isinstance(graph, DirectedGraph):
+        out_graph = nx.DiGraph()
+    else:
+        out_graph = nx.Graph()
+
+    if graph.is_spatial():
+        for node_id, pos in zip(graph.get_nodes(), graph.get_coordinates()):
+            out_graph.add_node(node_id, pos=pos)
+    else:
+        out_graph.add_nodes_from(graph.get_nodes())
+
+    edges = graph.get_edges()
+    if isinstance(edges, list) and len(edges) > 0:
+        edges = np.concatenate(edges, axis=0)
+
+    edges_as_tuples = list(map(tuple, edges))
+    out_graph.add_edges_from(edges_as_tuples)
+
+    return out_graph
+
+
+COMPATIPLE_CLASSES = {
+    BaseGraph: lambda x: x,
+    nx.Graph: from_networkx,
+    pd.DataFrame: lambda x: UndirectedGraph(coords=x),
+    np.ndarray: lambda x: UndirectedGraph(coords=x),
+}
+
+
+def to_napari_graph(graph: Any) -> BaseGraph:
+    """Generic function to convert "any" graph to a napari-graph.
+
+    Supported formats:
+        - NetworkX
```

Comment:
This is not an extensive list? But I also think the above functions are a little too haphazard — it's a bit presumptuous to assume that dataframes and arrays are just coordinates rather than adjacency matrices (for example). I would also add to the list here "- napari_graph.UndirectedGraph (pass through) - napari_graph.DirectedGraph (pass through)". I *might* suggest adding as a possible format a tuple of (scipy sparse matrix, numpy array) representing edges and coordinates. This would make it extremely easy to convert a skan skeleton to a napari graph. ;) And note that all scipy sparse matrix / sparse array formats can be quickly and easily converted to coo format which is basically our edge buffer. But this can come in a later PR, just noting down here since you already have weird array input formats 😅

Reply:
@jni, thanks for pointing this out. I'll comment this on it's own PR comment.